app/src/main/java/id/rizmaulana/covid19/ui/dailygraph/DailyGraphViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package id.rizmaulana.covid19.ui.dailygraph
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import id.rizmaulana.covid19.data.mapper.CovidDailyDataMapper
+import id.rizmaulana.covid19.data.repository.Repository
+import id.rizmaulana.covid19.ui.adapter.viewholders.DailyItem
+import id.rizmaulana.covid19.ui.base.BaseViewModel
+import id.rizmaulana.covid19.util.SingleLiveEvent
+import id.rizmaulana.covid19.util.rx.SchedulerProvider
+
+/**
+ * rizmaulana 16/03/20.
+ */
+class DailyGraphViewModel(
+    private val appRepository: Repository,
+    private val schedulerProvider: SchedulerProvider
+) : BaseViewModel() {
+    private val _toastMessage = SingleLiveEvent<String>()
+    val toastMessage: LiveData<String>
+        get() = _toastMessage
+
+    private val _dailyItems = MutableLiveData<List<DailyItem>>()
+    val dailyItems: LiveData<List<DailyItem>>
+        get() = _dailyItems
+
+    fun loadDailyData() {
+        /*Assume daily data just got fresh data from remote api on previous page
+          for UX Purpose, we directly load cache
+        */
+        _dailyItems.postValue(CovidDailyDataMapper.transform(appRepository.getCacheDaily().orEmpty()))
```

Comment:
This means it can be only refreshed if we go back to previous page right? What if we want to refresh it on the page? @rizmaulana

Reply:
agree, better to just fetch data from api, and you just need to call the repository

Reply:
What about use cache for first call and then user can pull data from api if they want to fresh data?